docs/config.md

```diff
@@ -199,21 +199,28 @@ Obviously, you would only set one line, don't copy them all in!
 ## Editing diffs
 
 The `ui.diff-editor` setting affects the tool used for editing diffs (e.g.
-`jj split`, `jj amend -i`). The default is `meld`. The left and right
-directories to diff are passed as the first and second argument respectively.
+`jj split`, `jj amend -i`). The default is `meld`.
+
+`jj` replaces the following arguments:
+
+- `$left` and `$right` are replaced with the paths to the left and right
+  directories to diff respectively.
+
+If no interpolation parameters are found, `$left $right` is appended.
 
 For example:
 
 ```toml
-ui.diff-editor = "kdiff3" # Use merge-tools.kdiff3.edit-args
-ui.diff-editor = ["kdiff3", "--merge"] # Specify edit-args inline
+# Use merge-tools.kdiff3.edit-args
+ui.diff-editor = "kdiff3"
+# Specify edit-args inline, $left $right will be added by default
+ui.diff-editor = ["kdiff3", "--merge"]
 ```
 
-Custom arguments can be added, and will be inserted before the paths to diff:
-
 ```toml
 # merge-tools.kdiff3.program = "kdiff3"      # Defaults to the name of the tool if not specified
-merge-tools.kdiff3.edit-args = ["--merge", "--cs", "CreateBakFiles=0"]
+merge-tools.kdiff3.edit-args = [
+    "--merge", "--cs", "CreateBakFiles=0", "$left", "$right"]
 ```
 
```

Comment:
I don't think it is documented that the setting and `ui.merge-editor` can be a either list or a space-separated string. We could add something here:

> The value of `ui.diff-editor` or `ui.TOOL.edit-args' can be either a string (arguments will be separated by spaces, there is no escaping) or a TOML list of strings.

I would also insert a similar sentence for merge arguments.

This is assuming that `ui.diff-editor` can be a space-separated string, I'm unsure if it's true. If not, that should be mentioned here as well.

Reply:
(I won't address it in this PR.)

Personally I want to discourage the use of space-separated string because the escape rule may be different from the platform behavior. (And we haven't implement it yet.)

Reply:
I think the space-separated string is a relatively new feature. I think I'd be OK with not allowing any escaping and forcing people to switch to lists if they want spaces inside their arguments.

In any case, it's the inconsistency between the different ways to specify the same option that bothers me. It's fine not to address it here.

Reply:
I didn't mean to disallow space-separated string, but to not advertise it in documentation. It's derived from `ui.editor` (and `$EDITOR`), so better to support this syntax for consistency.

Reply:
> It's derived from ui.editor (and $EDITOR), so better to support this syntax for consistency.

That's probably because I was new to TOML when I added that. I'm fine with dropping that support if we think that's better. I can see it being useful to use the simpler string format, but I also agree about the problems.